app/src/main/java/org/astraea/partitioner/smoothPartitioner/DependencyClient.java

```diff
@@ -1,6 +1,12 @@
 package org.astraea.partitioner.smoothPartitioner;
 
-// TODO
+/**
+ * The interface for the {@link SmoothWeightPartitioner}
+ *
+ * @see SmoothWeightPartitioner Enable users to send dependent data.
+ */
 public interface DependencyClient {
+  void startDependency();
```

Comment:
why we need this "method"? Users must call `startDependency` after getting `DependencyClient`, right? If so, the method is duplicate to the instantation of class.

Or the cost of creating `DependencyClient` is high and so you expect users should reuse `DependencyClient`?

Reply:
Because the method `dependencyClient()` to get `DependentClient` is static.If you want to put the operation to change to the start state in `dependencyClient()`, then the state we maintain must also be static. In this way, there will be thread-safe problems in this state under multiple producers.

If `dependencyClient()` is not static I can combine the two steps into one, but for this I have to ask the user to new a instance of Dependency.

For example:
```
pubuil class DependencyAdmin {
  dependencyClient()
}
*********************************
DpendencyAdmin dependencyAdmin = new DependencyAdmin();
dependencyAdmin.dependencyClient();
```

However, this also requires an additional operation...

Reply:
ok, please add comments to those methods.

Reply:
there is a major question. 

Why all `DependencyClient` share the same state (`dependencyManager.currentState`)? I feel you are complicating the story. You have to sync all operations between all `DependencyClient` when they are updating the state. However, the state of each `DependencyClient` should be isolated to each other.

Reply:
No, they don't share state.
I added a test to test to prove that the current DependencyClient does not affect each other.
>ok, please add comments to those methods.

```
pubuil class DependencyAdmin {
  dependencyClient()
}
*********************************
DpendencyAdmin dependencyAdmin = new DependencyAdmin();
dependencyAdmin.dependencyClient();
```

I want to make sure, I need to change this now, right.

Reply:
A partitioner has only one `DependencyManager`, right? If so, the multiples `DependencyClient` created by same `partitioner` will call same `DependencyManager`.

Reply:
>A partitioner has only one DependencyManager, right? If so, the multiples DependencyClient created by same partitioner will call same DependencyManager.

Sure,but the original intention of my design is that each producer will only maintain one state. In fact, I don't quite understand why a partitioner needs multiple DependencyManagers.